router/core/request_compression_test.go

```diff
@@ -0,0 +1,99 @@
+package core
+
+import (
+	"compress/flate"
+	"compress/gzip"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	br "github.com/andybalholm/brotli"
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+)
+
+func TestCompression(t *testing.T) {
+	httpRouter := chi.NewRouter()
+
+	// Adds Brotli compressor
+	brCompressor := middleware.NewCompressor(5, CustomCompressibleContentTypes...)
+	brCompressor.SetEncoder("br", func(w io.Writer, level int) io.Writer {
+		return br.NewWriterLevel(w, level)
+	})
+	httpRouter.Use(brCompressor.Handler)
+
+	// Adds deflate & gzip compressor
+	httpRouter.Use(middleware.AllowContentEncoding("deflate", "gzip"))
```

Comment:
I think it would makes sense, if we would validate all possible content types `"deflate", "gzip", "br"` before we add the compression middleware

Reply:
As you instructed, I have added code to validate all content types.

Reply:
Have I misunderstood?